src/ColumnUtils.js

```diff
@@ -8,6 +8,15 @@ module.exports = {
     }
   },
 
+  updateMetrics(metrics, idx, column){
+    if(Array.isArray(metrics.columns)){
+      metrics.columns.splice(idx, 1, column);
+    }else if (typeof Immutable !== 'undefined') {
```

Comment:
this is fine for now - but we should set a flag when the grid is initialised (most probably on the column metrics object) to say whether we are using immutable data or not